rules-reviewed/camel3/camel2/xml-moved-components.windup.groovy

```diff
@@ -26,13 +27,67 @@ import java.util.function.Function
 
 final String FROM_XML_FILES_IN_PROJECT = "xmlFilesInProject"
 final String FROM_FILES_IN_PROJECT = "filesInProject"
+
+/*
+returns xpath condition that counts elements
+ */
+ String countXpath(String... elements) {
+    StringBuilder sb = new StringBuilder("//*[( ")
+    for (i = 0; i < elements.length; i++) {
+        sb.append("count(c:" + elements[i] + ")")
+        if (i < (elements.length - 1)) {
+            sb.append(" + ")
+        }
+    }
+    sb.append(")> 0]")
+    return sb.toString()
+}
+
+static String componentNameExpression(String componentName) {
+    return "(\"$componentName:"
+}
+
+static String componentNameXpath(String component) {
+    return "//*/c:route/*[starts-with(@uri, '$component:')]"
+}
+
+final Map<String, String> movedComponents = [
+        // component name: [xml xpath, java matches expression ]
+        "language"     : [componentNameXpath("language"), componentNameExpression("language")],
```

Comment:
I know Groovy swallows this, but, you are declaring a Map<String, String[] or List<String>> but you have defined movedComponents as a Map<String,String> ..... is that correct ?

Reply:
Correct, thanks! Should be fixed now.